src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -0,0 +1,280 @@
+        private const string DurableTask = "durableTask";
+
+        private readonly IOptionsMonitor<ScriptApplicationHostOptions> _applicationHostOptions;
+        private readonly ILogger _logger;
+        private readonly IEnumerable<WorkerConfig> _workerConfigs;
+        private readonly HttpClient _httpClient;
+        private readonly ISecretManagerProvider _secretManagerProvider;
+        private readonly IConfiguration _configuration;
+        private readonly IHostIdProvider _hostIdProvider;
+
+        private CloudBlockBlob _hashBlob;
+
+        public FunctionsSyncManager(IConfiguration configuration, IHostIdProvider hostIdProvider, IOptionsMonitor<ScriptApplicationHostOptions> applicationHostOptions, IOptions<LanguageWorkerOptions> languageWorkerOptions, ILoggerFactory loggerFactory, HttpClient httpClient, ISecretManagerProvider secretManagerProvider)
+        {
+            _applicationHostOptions = applicationHostOptions;
+            _logger = loggerFactory?.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+            _workerConfigs = languageWorkerOptions.Value.WorkerConfigs;
+            _httpClient = httpClient;
+            _secretManagerProvider = secretManagerProvider;
+            _configuration = configuration;
+            _hostIdProvider = hostIdProvider;
+        }
+
+        private CloudBlockBlob HashBlob
+        {
+            get
+            {
+                if (_hashBlob == null)
+                {
+                    string storageConnectionString = _configuration.GetWebJobsConnectionString(ConnectionStringNames.Storage);
+                    CloudStorageAccount account = null;
+                    if (!string.IsNullOrEmpty(storageConnectionString) &&
+                        CloudStorageAccount.TryParse(storageConnectionString, out account))
+                    {
+                        string hostId = _hostIdProvider.GetHostIdAsync(CancellationToken.None).GetAwaiter().GetResult();
+                        CloudBlobClient blobClient = account.CreateCloudBlobClient();
+                        var blobContainer = blobClient.GetContainerReference(ScriptConstants.AzureWebJobsHostsContainerName);
+                        string hashBlobPath = $"synctriggers/{hostId}/last";
+                        _hashBlob = blobContainer.GetBlockBlobReference(hashBlobPath);
+                    }
+                }
+                return _hashBlob;
+            }
+        }
+
+        public async Task<(bool success, string error)> TrySyncTriggersAsync(bool checkHash = false)
+        {
+            var payload = await GetSyncTriggersPayload();
+            string json = JsonConvert.SerializeObject(payload);
+
+            bool shouldSyncTriggers = true;
+            if (checkHash)
+            {
+                shouldSyncTriggers = await CheckAndUpdateHashAsync(json);
+            }
+
+            if (shouldSyncTriggers)
+            {
+                return await SetTriggersAsync(json);
```

Comment:
With this flow, if the `SetTriggersAsync` call fails, we'll be left with an updated hash even if the call was not made. Ideally, the hash should only be updated if this call succeeds.